File: PenetrationProjectAPI/src/main/java/com/ssafy/cafe/model/dto/Product.java
package com.ssafy.cafe.model.dto;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Product {

	private Integer id;
	private String name;
	private String type;
	private Integer price;
	private String img;
	private String filename;
	private String filepath;

	@Builder
	public Product(Integer id, String name, String type, Integer price, String img) {
		super();
		this.id = id;
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
	}

	public Product(String name, String type, Integer price, String img) {
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
	}

	public Product(Integer id, String name, String type, Integer price, String img, String filename, String filepath) {
		super();
		this.id = id;
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
		this.filename = filename;
		this.filepath = filepath;
	}

	public Product(String name, String type, Integer price, String img, String filename, String filepath) {
		super();
		this.name = name;
		this.type = type;
		this.price = price;
		this.img = img;
		this.filename = filename;
		this.filepath = filepath;
	}
}

File: admin_final_gumi0607_09/app/src/main/java/com/ssafy/admin_final_gumi0607_09/data/remote/dto/Product.kt
package com.ssafy.admin_final_gumi0607_09.data.remote.dto

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import java.io.Serializable

@Parcelize
data class Product(
    val name: String,
    val type: String,
    val price: Int,
    val img: String
):Parcelable{

    @SerializedName("id")
    var id: Int = -1

    constructor(id: Int, name: String, type: String, price: Int, img: String): this(name, type, price, img){
        this.id =id
    }
}
